### PR TITLE
Add gcc11 cuda11.8 docker image

### PR DESCRIPTION
**Pull request by ezyang**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #106804

Signed-off-by: Edward Z. Yang <ezyang@meta.com>

### CODE REVIEW

**Comment by ezyang** [comment]
@seemethere you are probably not the right reviewer but maybe you know who the right reviewer is

**Review by seemethere — APPROVED** [review]

File: .ci/docker/build.sh [hunk 1]
```diff
@@ -160,6 +160,35 @@ case "$image" in
...
+    PROTOBUF=yes
+    DB=yes
+    VISION=yes
+    KATEX=yes
+    UCX_COMMIT=${_UCX_COMMIT}
+    UCC_COMMIT=${_UCC_COMMIT}
+    CONDA_CMAKE=yes
+    TRITON=yes
+    ;;
+    pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc11-inductor-benchmarks)
```
Comment (huydhn) [inline .ci/docker/build.sh:177]: Nit: `pytorch-linux-bionic-cuda11.8-cudnn8-py3-gcc11-inductor-benchmarks` has a wrong alignment

**Review by huydhn — APPROVED** [review]
LGTM

**Comment by ezyang** [comment]
https://github.com/pytorch/pytorch/pull/105260 is a better, more comprehensive version of this

**Pull request closed without merging**